core/src/main/java/org/elasticsearch/monitor/jvm/JvmInfo.java

```diff
@@ -114,15 +114,33 @@
             Class<?> vmOptionClazz = Class.forName("com.sun.management.VMOption");
             PlatformManagedObject hotSpotDiagnosticMXBean = ManagementFactory.getPlatformMXBean(clazz);
             Method vmOptionMethod = clazz.getMethod("getVMOption", String.class);
-            Object useCompressedOopsVmOption = vmOptionMethod.invoke(hotSpotDiagnosticMXBean, "UseCompressedOops");
             Method valueMethod = vmOptionClazz.getMethod("getValue");
-            info.useCompressedOops = (String)valueMethod.invoke(useCompressedOopsVmOption);
-            Object useG1GCVmOption = vmOptionMethod.invoke(hotSpotDiagnosticMXBean, "UseG1GC");
-            info.useG1GC = (String)valueMethod.invoke(useG1GCVmOption);
-        } catch (Throwable t) {
-            // unable to deduce the state of compressed oops
-            info.useCompressedOops = "unknown";
-            info.useG1GC = "unknown";
+
+            try {
+                Object useCompressedOopsVmOption = vmOptionMethod.invoke(hotSpotDiagnosticMXBean, "UseCompressedOops");
+                info.useCompressedOops = (String) valueMethod.invoke(useCompressedOopsVmOption);
+            } catch (Exception ignored) {
+            }
+
+            try {
+                Object useG1GCVmOption = vmOptionMethod.invoke(hotSpotDiagnosticMXBean, "UseG1GC");
+                info.useG1GC = (String) valueMethod.invoke(useG1GCVmOption);
+            } catch (Exception ignored) {
+            }
+
+            try {
+                Object initialHeapSizeVmOption = vmOptionMethod.invoke(hotSpotDiagnosticMXBean, "InitialHeapSize");
+                info.configuredInitialHeapSize = Long.parseLong((String) valueMethod.invoke(initialHeapSizeVmOption));
+            } catch (Exception ignored) {
+            }
+
+            try {
+                Object maxHeapSizeVmOption = vmOptionMethod.invoke(hotSpotDiagnosticMXBean, "MaxHeapSize");
+                info.configuredMaxHeapSize = Long.parseLong((String) valueMethod.invoke(maxHeapSizeVmOption));
+            } catch (Exception ignored) {
```

Comment:
@nik9000 By the time we get that far in that block, nothing should happen there and those try/catch blocks are defensive more than anything. The expected way to see an exception out of that entire block is if the end user is not on the HotSpot VM or `com.sun.management.HotSpotDiagnosticMXBean` is otherwise unavailable.

I do not think that we should log in that case as there is nothing the user can actively do other than switch to HotSpot. But we don't support non-HotSpot. :smile:

What I *do* think that we should do though is write a test that tests that these methods do work on HotSpot because otherwise they can silently be broken and we will never no. I will take that on.